backend/src/main/java/com/example/cms/user/UserService.java

```diff
@@ -127,25 +135,45 @@ private void validatePassword(String password) {
         }
     }
 
-    @Secured("ROLE_USER")
-    public UserDtoDetailed updateUser(Long id, UserDtoFormUpdate form) {
+    private User validateUserAndForm(Long id, UserDtoFormUpdate form) {
         User user = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+
         if (securityService.isForbiddenUser(user)) {
             throw new UserForbiddenException();
         }
 
-        if (userRepository.existsByUsername(form.getUsername())) {
+        if (userRepository.existsByUsername(form.getUsername())
+                && !user.getUsername().equals(form.getUsername())) {
```

Comment:
Rozwiązanie problemu z https://github.com/AdrianKokot/put-reunice-platform/issues/139

Reply:
Wydaje się, że powinieneś sprawdzać ID aktualnie zalogowanego użytkownika i ID tego znalezionego. W tym momencie możesz mieć sytuację, że edytujesz innego użytkownika, ustawiasz mu login taki sam jak masz Ty i to przejdzie.
Nie wiem po co by miał ktoś chcieć coś takiego zrobić ale nie powinno się dać tak zrobić.

Jeśli coś przeoczyłem i nie da się tak zrobić jak napisałem to jest git

Reply:
To w jakiś sposób, chyba wg twoich wskazówek poprawiłem, na pozostałe robię issue do rozważenia na przyszłość